bin/schema_generator.sh

```diff
@@ -0,0 +1,156 @@
+
+    mkdir -p /app/.ssh
+
+    echo "$MPS_SSH_KEY_BASE64" | base64 --decode > /app/.ssh/id_ed25519
+    ssh-keyscan github.com > /app/.ssh/known_hosts # Makes the future git-push non-interactive
+
+    chown -R "$(id -u):$(id -g)" "$HOME/.ssh"
+    chmod 700 "$HOME/.ssh"
+    chmod 700 "$HOME/.ssh/id_ed25519"
+}
+
+function setup_dependencies() {
+    # Installs mozilla-schema-generator in a virtual environment
+
+    virtualenv msg-venv
+    # shellcheck disable=SC1091
+    source msg-venv/bin/activate
+    pip install -e ./mozilla-schema-generator
+}
+
+function clone_and_configure_mps() {
+    # Checkout mozilla-pipeline-schemas and changes directory to prepare for
+    # schema generation.
+
+    rm -rf mozilla-pipeline-schemas
+
+    git clone git@github.com:mozilla-services/mozilla-pipeline-schemas.git
+    cd mozilla-pipeline-schemas/$MPS_SCHEMAS_DIR || exit
+    git checkout $MPS_BRANCH_SOURCE
+
+    git branch -D $MPS_BRANCH_WORKING
+    git checkout -b $MPS_BRANCH_WORKING
+}
+
+function prepare_metadata() {
+    # Assumes that all schemas under the current directory are valid JSON schema.
+
+    local telemetry_metadata="metadata/telemetry-ingestion/telemetry-ingestion.1.schema.json"
+    local structured_metadata="metadata/structured-ingestion/structured-ingestion.1.schema.json"
+
+    find ./telemetry -type f -exec metadata_merge $telemetry_metadata {} ";"
+    find . -path ./telemetry -prune -o -type f -exec metadata_merge $structured_metadata {} ";"
+}
+
+function filter_schemas() {
+    # Remove metadata
+    rm -rf metadata
+
+    # Pioneer-study is not nested, remove it
+    rm -rf pioneer-study
+
+    # Replace newlines with backticks (hard to do with sed): cat | tr
+    # Remove the last backtick; it's the file-ending newline: rev | cut | rev
+    # Replace backticks with "\|" (can't do that with tr): sed
+    # Find directories that don't match any of the regex expressions: find
+    # Remove them: rm
+    < /app/mozilla-schema-generator/bin/allowlist tr '\n' '`' | \
+        rev | cut -c 2- | rev | \
+        sed -e 's/`/\\\\|/g' | \
+        xargs -I % find . -type f -regextype sed -not -regex '.*/\(%\|metadata/\)/.*' | grep ".bq" | \
+        xargs rm -rf
+}
+
+function commit_schemas() {
+    # This method will keep a changelog of releases. If we delete and newly
+    # checkout branches everytime, that will contain a changelog of changes.
+    # Assumes the current directory is the root of the repository
+
+    find . -name "*.bq" -type f -exec git add {} +
+    git checkout ./*.schema.json
+    git commit -a -m "Interim Commit"
+
+    git checkout $MPS_BRANCH_PUBLISH || git checkout -b $MPS_BRANCH_PUBLISH
+
+    # Keep only the schemas dir
+    find .  -mindepth 1 -maxdepth 1 -not -name .git -exec rm -rf {} +
+    git checkout $MPS_BRANCH_WORKING -- schemas
+    git commit -a -m "Auto-push from schema generation"
+}
+
+function main() {
+    cd $BASE_DIR || exit
+
+    # -1. Setup ssh key and git config
```

Comment:
Why is this -1?

Reply:
An original comment, I removed the numbering since the functions better define the scope of work.